chapters/VK_EXT_present_timing/PresentTimeInfo.adoc

```diff
@@ -0,0 +1,161 @@
+
+  * pname:targetPresentTime, if non-zero, specifies the earliest time the
+    application wants the image to complete the swapchain's target present
+    stage. A value of zero specifies that the presentation engine may:
+    display the image at any time allowed by the current present mode.
+  * pname:minPresentDuration, if non-zero, specifies the minimum duration in
+    nanoseconds before which the next image presentation request can: reach
+    the target present stage. A value of zero specifies that the
+    presentation engine may: display the image for any duration allowed by
+    the current present mode.
+
+The contents of the sname:VkPresentTimeEXT union is interpreted depending on
+the associated sname:VkPresentInfoKHR::pname:pSwapchains member creation
+values. If
+sname:VkSwapchainPresentTimingCreateInfoEXT::pname:presentAtRelativeTime was
+ename:VK_FALSE when creating the pname:swapchain, pname:targetPresentTime is
+used. Otherwise, pname:minPresentDuration is used.
+
+pname:targetPresentTime is a time in nanoseconds, according to the
+time-domain being used. The presentation engine must: not complete the
+target present stage, specified in the associated swapchain
+slink:VkSwapchainPresentTimingCreateInfoEXT::pname:presentStageTarget, at an
+earlier time.
+
+[NOTE]
+.Note
+====
+Applications that use pname:targetPresentTime to specify an absolute present
+time should: regularly rebase their calculations for their next target time
+on the feedback from flink:vkGetPastPresentationTimingEXT to compensate for
+accumulated precision errors or potential clock drift. It is recommended
+that when targeting the time of a vertical blanking period, applications
+subtract a small delta from pname:targetPresentTime as a margin for error.
+====
+
+include::{generated}/validity/structs/VkPresentTimeEXT.adoc[]
+--
+
+[open,refpage='VkPresentStageTimeEXT',desc='Associate a present stage with a timestamp',type='structs']
+--
+The sname:VkPresentStageTimeEXT structure is defined as:
+
+include::{generated}/api/structs/VkPresentStageTimeEXT.adoc[]
+
+  * pname:stage is a tlink:VkPresentStageFlagsEXT value specifying a present stage.
+  * pname:time is a time in nanoseconds associated with the pname:stage.
+
+include::{generated}/validity/structs/VkPresentStageTimeEXT.adoc[]
+--
+
+[open,refpage='VkPresentStageFlagBitsEXT',desc='Bitmask specifying stages of the image presentation process',type='enums']
+--
+
+Presenting an image to the user typically involves multiple stages. Bits
+which can: be set to specify present stages are:
+
+include::{generated}/api/enums/VkPresentStageFlagBitsEXT.adoc[]
+
+  * ename:VK_PRESENT_STAGE_IMAGE_HANDOFF_BIT_EXT is the stage after which
+    the presentation request of the image has been passed to a
+    system-specific entity of the presentation engine that is outside the
+    control of the Vulkan implementation. This stage is the only
+    ename:VkPresentStageFlagBitsEXT that is required: to be supported.
```

Comment:
Does `VK_PRESENT_STAGE_IMAGE_HANDOFF_BIT_EXT` provide a timestamp equivalent to `presentReadyTime`? Doing so would require waiting for GPU sync specified in the present, but the description of `VK_PRESENT_STAGE_IMAGE_HANDOFF_BIT_EXT` does not seem to imply it.

Reply:
My interpretation of VK_PRESENT_STAGE_IMAGE_HANDOFF_BIT_EXT is that the “wait semaphores” have signaled (i.e. rendering is done).  That is what the VK_GOOGLE_display_timing Android implementation uses.  Is that the intention of VK_PRESENT_STAGE_IMAGE_HANDOFF_BIT_EXT?